src/backend/opencl/blas.cpp

```diff
@@ -113,6 +118,10 @@ template<typename T>
 Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs,
                 af_mat_prop optLhs, af_mat_prop optRhs)
 {
+    if(OpenCLCPUOffload()) {
```

Comment:
Wouldn't this change the behavior of the function. The CPU version does not seem to be asynchronous.